packages/framework/src/Component/Domain/Domain.php

```diff
@@ -20,26 +21,35 @@ class Domain implements DomainIdsProviderInterface
     /**
      * @var \Shopsys\FrameworkBundle\Component\Domain\Config\DomainConfig|null
      */
-    protected $currentDomainConfig;
+    protected ?DomainConfig $currentDomainConfig = null;
 
     /**
-     * @var \Shopsys\FrameworkBundle\Component\Domain\Config\DomainConfig[]
+     * @var bool
      */
-    protected $domainConfigs;
+    protected bool $isOnCdnDomain;
 
     /**
-     * @var \Shopsys\FrameworkBundle\Component\Setting\Setting
+     * @var string|null
      */
-    protected $setting;
+    protected ?string $cdnDomain = null;
 
     /**
      * @param \Shopsys\FrameworkBundle\Component\Domain\Config\DomainConfig[] $domainConfigs
      * @param \Shopsys\FrameworkBundle\Component\Setting\Setting $setting
      */
-    public function __construct(array $domainConfigs, Setting $setting)
+    public function __construct(
+        protected array $domainConfigs,
+        protected Setting $setting,
+    ) {
+        $this->isOnCdnDomain = false;
+    }
```

Comment:
I think this behavior around CDN domain should not be necessary here in Domain, as it is not really a domain. It should be moved somewhere else.